File: Decorator/Java/CondimentDecorator.java
package com.company;

abstract public class CondimentDecorator implements Beverage {
    private Beverage base;

    public CondimentDecorator(Beverage base) {
        super();
        this.base = base;
    }

    @Override
    public int getTotalPrice() {
        return base.getTotalPrice();
    }

    @Override
    public String getDescription() {
        return base.getDescription();
    }

    protected Beverage getBase() {
        return base;
    }
}

File: Decorator/Kotlin/CondimentDecorator.kt
open abstract class CondimentDecorator(b: Beverage) : Beverage by b {
    private var base: Beverage = b

    override fun getTotalPrice(): Int {
        return base.getTotalPrice()
    }

    override fun getDescription(): String {
        return base.getDescription()
    }

    protected fun getBase() : Beverage {
        return base
    }
}